exporter/datasetexporter/logs_exporter.go

```diff
@@ -65,6 +66,73 @@ func buildBody(attrs map[string]interface{}, value pcommon.Value) string {
 	return message
 }
 
+// Function maps OTel severity on the LogRecord to DataSet severity level (number)
+func otelSeverityToDataSetSeverity(log plog.LogRecord) int {
+	// If log record doesn't contain severity or we can't map it to a valid DataSet severity,
+	// we use this value (INFO) instead
+	defaultSeverityLevel := 3
+
+	// This function maps OTel severity level to DataSet severity levels
+	//
+	// Valid OTel levels - https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber
+	// and valid DataSet ones - https://github.com/scalyr/logstash-output-scalyr/blob/master/lib/logstash/outputs/scalyr.rb#L70
+	sevNum := log.SeverityNumber()
+	sevText := log.SeverityText()
+
+	dataSetSeverity := defaultSeverityLevel
+
+	if sevNum > 0 {
+		// See https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber
+		// for OTEL mappings
+		switch sevNum {
+		case 1, 2, 3, 4:
+			// TRACE
+			dataSetSeverity = 1
+		case 5, 6, 7, 8:
+			// DEBUG
+			dataSetSeverity = 2
+		case 9, 10, 11, 12:
+			// INFO
+			dataSetSeverity = 3
+		case 13, 14, 15, 16:
+			// WARN
+			dataSetSeverity = 4
+		case 17, 18, 19, 20:
+			// ERROR
+			dataSetSeverity = 5
+		case 21, 22, 23, 24:
+			// FATAL / CRITICAL / EMERGENCY
+			dataSetSeverity = 6
+		}
+	} else if sevText != "" {
+		// Per docs, SeverityNumber is optional so if it's not present we fall back to SeverityText
+		// https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitytext
+		switch strings.ToLower(sevText) {
+		case "fine", "finest":
+			dataSetSeverity = 0
```

Comment:
Consider defining DataSet severities as constants for better readability. It also may be reused in tests.